src/utils/requestUtil.ts

```diff
@@ -4,16 +4,37 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { WebResource } from "ms-rest";
+import { ServiceClientCredentials } from "ms-rest";
 import * as request from 'request-promise';
 import { appendExtensionUserAgent } from "vscode-azureextensionui";
+import { IFunctionHostKeyResponse } from "../models/Function";
+import { signRequest } from "./signRequest";
 
-export async function requestUtil(url: string) : Promise<string> {
+export type nRequest = WebResource & request.RequestPromiseOptions;
+
+export async function requestUtil(url: string): Promise<string> {
     const requestOptions: WebResource = new WebResource();
     requestOptions.headers = {
         ['User-Agent']: appendExtensionUserAgent()
     };
     requestOptions.url = url;
-// tslint:disable-next-line: await-promise
+    // tslint:disable-next-line: await-promise
     const response = await request(requestOptions).promise();
-    return  <string>(response);
+    return <string>(response);
+}
+
+export async function requestUtilWithCredentials(url: string, credentials: ServiceClientCredentials): Promise<string> {
+    const requestOptions: WebResource = new WebResource();
+    requestOptions.headers = {
+        ['User-Agent']: appendExtensionUserAgent()
+    };
+    requestOptions.url = url;
+    await signRequest(requestOptions, credentials);
+    // tslint:disable-next-line: await-promise
+    const response = await request(requestOptions).promise();
+    return <string>(response);
+}
+
+export async function sendRequest(httpReq: nRequest): Promise<IFunctionHostKeyResponse> {
```

Comment:
could we use Promise<T> instead of Promise<IFunctionHostKeyResponse>

Reply:
I add this function instead of "requestUtil" due to the limitation of of passing body json which cause the api can't recognize it. I haven't come up with a better way. And this is how azure function's extension deals with it.

Reply:
instead of IFunctionHostKeyResponse can this be a generic <T>